engine/src/main/java/org/terasology/i18n/TranslationSystem.java

```diff
@@ -39,18 +26,31 @@
 
     /**
      * If no perfect match is found for the default locale, fallback strategies will attempt to find the closest match.
-     * @param id must match the pattern <code>${module:project#id}</code>
+     * @param id must match the pattern {@code ${module:project#id}}
+     * @param arguments {@link java.text.MessageFormat} arguments for the translation string
      * @return the translated string
      */
-    String translate(String id);
+    String translate(String id, Object... arguments);
 
     /**
      * If no perfect match is found for the given locale, fallback strategies will attempt to find the closest match.
-     * @param id must match the pattern <code>${module:project#id}</code>
+     * @param id must match the pattern {@code ${module:project#id}}
+     *
      * @param locale the target locale
+     * @param arguments  {@link java.text.MessageFormat} arguments for the translation string
      * @return the translated string
      */
-    String translate(String id, Locale locale);
+    String translate(String id, Locale locale, Object... arguments);
+
+    @Override
+    default String translate(String id) {
+        return translate(id, new Object[0]);
+    }
+
+    @Override
+    default String translate(String id, Locale locale) {
+        return translate(id, locale, new Object[0]);
+    }
```

Comment:
After the NUI Translate interface was moved to an external library, I had to add these default implementations that delegate to our varargs versions.

It's not a great look.

Reply:
That's not terrible :) Does that mean the third checkbox can be checked? Since we're now post-NUI extraction